sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/LegacyRestClient.java

```diff
@@ -60,20 +51,24 @@ public String post(String url, String body, String contentType) {
         String result = null;
         try (CloseableHttpClient client = HttpClients.createDefault()) {
             HttpPost httpPost = new HttpPost(url);
-            httpPost.setEntity(HttpEntities.create(body, ContentType.create(contentType)));
-            HttpClientResponseHandler<String> responseHandler = (ClassicHttpResponse response) -> {
-                int status = response.getCode();
-                String result1 = null;
-                if (status >= 200 && status < 300) {
-                    HttpEntity entity = response.getEntity();
-                    result1 = entity != null ? EntityUtils.toString(entity) : null;
-                }
-                return result1;
-            };
-            result = client.execute(httpPost, responseHandler);
+            httpPost.setEntity(
+                new StringEntity(body, ContentType.create(contentType)));
+            result = client.execute(httpPost, responseHandler());
         } catch (IOException ioe) {
             ioe.printStackTrace();
         }
         return result;
     }
+
+    private ResponseHandler<String> responseHandler() {
```

Comment:
Shall we change the function name to "createResponseHandler"

Reply:
ok, it's easier to understand.